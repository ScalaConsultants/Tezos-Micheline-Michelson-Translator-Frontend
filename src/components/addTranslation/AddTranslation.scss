.AddTranslation {
    .add-translation_code-displays {
        display: flex;
        justify-content: space-between;
        margin-left: auto;
        margin-right: auto;
        width: 80%;

        .form-code-display {
            width: 45%;

            .form-code-display_value {
                max-height: 200px;
            }
        }

        @media screen and (max-width: 500px) {
            display: block;

            .form-code-display {
                width: 100%;
            }
        }
    }

    .add-translation_fields-line {
        display: flex;
        width: 80%;
        margin-left: auto;
        margin-right: auto;
        justify-content: space-between;

        .form-input {
            width: 45%;
        }

        @media screen and (max-width: 500px) {
            display: block;

            .form-input {
                width: 100%;
            }
        }
    }

    .form-textarea {
        width: 80%;
        margin-left: auto;
        margin-right: auto;
        margin-bottom: 10px;
    }

    .add-translation_buttons {
        display: flex;
        width: 220px;
        justify-content: space-between;
        margin-left: auto;
        margin-right: auto;
        margin-top: 24px;
        margin-bottom: 16px;
    }
}
