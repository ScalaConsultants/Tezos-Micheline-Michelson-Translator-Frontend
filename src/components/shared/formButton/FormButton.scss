@import "../../../scss/vars";

.form-button {
    font-weight: 500;
    text-transform: uppercase;
    letter-spacing: 0.15;
    font-size: 14px;
    box-shadow: $box-shadow;
    border-radius: 4px;
    padding: 5px;
    height: 40px;
    min-width: 100px;
    border: none;
    color: white;

    &:hover {
        cursor: pointer;
    }
}

.form-button_submit {
    background: $background-color-red;
}

.form-button_secondary {
    background: $background-color-grey-3;
}
