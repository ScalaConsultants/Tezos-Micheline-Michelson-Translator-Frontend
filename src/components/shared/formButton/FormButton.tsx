import React from "react";
import { ButtonProps } from "./types";

const FormButton = ({ type, label, className, disabled, stylingType, onClick }: ButtonProps) => (
  <button
    className={`form-button form-button_${stylingType || "submit"} ${className}`}
    data-testid="form-button"
    disabled={disabled}
    type={type}
    onClick={onClick}
  >
    {label}
  </button>
);

export default FormButton;
