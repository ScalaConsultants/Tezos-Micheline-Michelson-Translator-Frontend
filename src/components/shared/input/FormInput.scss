@import "../../../scss/vars";

.form-input {
    .form-input_input {
        display: block;
        min-width: 200px;
        width: 90%;
        border: none;
        border-bottom: solid 2px $border-red;
        font-size: 16px;
        padding-bottom: 10px;
        caret-color: $red;

        &:focus {
            outline: none;
        }
    }

    .form-input_label {
        text-align: left;
        color: $text-red;
        font-size: 12px;
        margin-bottom: 5px;
    }

    .form-input_validation-marker {
        border-radius: 100%;
        border: 2px solid $border-black;
        opacity: 0.6;
        width: 8px;
        height: 8px;
        position: relative;
        bottom: 30px;
        left: 85%;

        &.valid {
            background: $background-color-black;
        }
    }

    .form-input_error {
        height: 20px;
        text-align: left;
        opacity: 0.6;
        font-size: 12px;
        margin-top: -10px;
    }
}
