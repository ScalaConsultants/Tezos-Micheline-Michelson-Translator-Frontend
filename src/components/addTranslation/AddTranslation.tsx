import React from "react";
import { Formik } from "formik";
import * as Yup from "yup";
import { useMappedState } from "redux-react-hook";
import FormTextarea from "../shared/textarea/FormTextarea";
import FormInput from "../shared/input/FormInput";
import { IState } from "../../store/translator/types";
import FormCodeDisplay from "../shared/formCodeDisplay/FormCodeDisplay";
import FormButton from "../shared/formButton/FormButton";
import { AddTranslationState, FormValues } from "./types";
import "./AddTranslation.scss";

const mapState = (state: IState) => ({
  translator: state.translator,
});

const validationSchema = Yup.object().shape({
  name: Yup.string(),
  author: Yup.string(),
  description: Yup.string(),
});

const AddTranslation = ({ setShowModal }: AddTranslationState) => {
  const { translator } = useMappedState(mapState);

  const submitForm = (values: FormValues) => {
    setShowModal(false);
  };

  return (
    <div className="AddTranslation">
      <Formik
        initialValues={{
          name: "",
          author: "",
          description: "",
        }}
        validationSchema={validationSchema}
        onSubmit={submitForm}
      >
        {({ values, errors, touched, handleChange, handleBlur, handleSubmit, isSubmitting }) => (
          <form onSubmit={handleSubmit} noValidate>
            <div className="add-translation_fields-line">
              <FormInput
                label="Name"
                type="text"
                name="name"
                onChange={handleChange}
                onBlur={handleBlur}
                value={values.name}
                errors={errors.name}
                touched={touched.name}
              />
              <FormInput
                label="Author"
                type="text"
                name="author"
                onChange={handleChange}
                onBlur={handleBlur}
                value={values.author}
                errors={errors.author}
                touched={touched.author}
              />
            </div>
            <FormTextarea
              label="Description"
              placeholder="Fill in the description."
              name="description"
              onChange={handleChange}
              onBlur={handleBlur}
              value={values.description}
              errors={errors.description}
              touched={touched.description}
            />
            <div className="add-translation_code-displays">
              <FormCodeDisplay value={translator.micheline} type="Micheline" />
              <FormCodeDisplay value={translator.michelson} type="Michelson" />
            </div>
            <div className="add-translation_buttons">
              <FormButton label="cancel" type="secondary" onClick={() => setShowModal(false)} />
              <FormButton label="save" type="submit" onClick={() => submitForm(values)} disabled={isSubmitting} />
            </div>
          </form>
        )}
      </Formik>
    </div>
  );
};

export default AddTranslation;
