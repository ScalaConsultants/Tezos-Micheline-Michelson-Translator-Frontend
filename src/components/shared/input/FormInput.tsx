import React from "react";
import "./FormInput.scss";
import { FormInputProps } from "./types";

const FormInput = ({ label, name, type, onChange, onBlur, value, errors, touched, className }: FormInputProps) => (
  <div className={`form-input ${className}`}>
    <p className="form-input_label">{label}</p>
    <input
      name={name}
      type={type}
      onChange={onChange}
      onBlur={onBlur}
      placeholder={label}
      value={value}
      className="form-input_input"
    />
    <div className={`form-input_validation-marker ${touched && !errors && "valid"}`} />
    <div className="form-input_error">{errors && touched && errors}</div>
  </div>
);

export default FormInput;
