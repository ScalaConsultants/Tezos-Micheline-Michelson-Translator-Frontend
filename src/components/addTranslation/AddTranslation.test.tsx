import React from "react";
import { mount, ReactWrapper } from "enzyme";
import { StoreContext } from "redux-react-hook";
import configureStore from "redux-mock-store";
import AddTranslation from "./AddTranslation";
import FormButton from "../shared/formButton/FormButton";
import FormInput from "../shared/input/FormInput";

const translator = {
  title: "Title1",
  micheline:
    "parameter int;\nstorage int;\ncode { CAR ;\n       PUSH int 1 ;\n       ADD ;\n       NIL operation ;\n       PAIR }",
  michelson:
    '[ { "prim": "parameter", "args": [ { "prim": "int" } ] }, { "prim": "storage", "args": [ { "prim": "int" } ] }, { "prim": "code", "args": [ [ { "prim": "CAR" },{ "prim": "PUSH", "args": [ { "prim": "int" }, { "int": "1" } ] },{ "prim": "ADD" },{ "prim": "NIL", "args": [ { "prim": "operation" } ] },{ "prim": "PAIR" } ] ] } ]\n',
  error: "",
  isErrorOrEmpty: true
};

const mockStore = configureStore();

describe("AddTranslation", () => {
  let wrapper: ReactWrapper;
  let store: any;

  beforeEach(() => {
    store = mockStore({ translator });

    store.dispatch = jest.fn();

    wrapper = mount(
      <StoreContext.Provider value={store}>
        <AddTranslation setShowModal={() => {}} />
      </StoreContext.Provider>,
    );
  });

  it("renders without crashing", () => {
    const item = wrapper.find(AddTranslation);

    expect(
      item
        .at(0)
        .find(FormInput)
        .at(0)
        .text(),
    ).toEqual("Title");

    expect(
      item
        .at(0)
        .find(FormButton)
        .at(0)
        .text(),
    ).toEqual("close");
  });

  it("matches snapshot", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

});
