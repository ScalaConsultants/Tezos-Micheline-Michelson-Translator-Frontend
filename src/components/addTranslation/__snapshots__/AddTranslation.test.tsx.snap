// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTranslation matches snapshot 1`] = `
"<AddTranslation setShowModal={[Function: setShowModal]}>
  <div className=\\"add-translation\\">
    <Formik initialValues={{...}} validationSchema={{...}} onSubmit={[Function: submitForm]}>
      <form onSubmit={[Function]} noValidate={true}>
        <div className=\\"add-translation_fields-line\\">
          <FormInput label=\\"Title\\" type=\\"text\\" name=\\"title\\" onChange={[Function]} onBlur={[Function]} value=\\"\\">
            <div className=\\"form-input undefined\\">
              <p className=\\"form-input_label\\">
                Title
              </p>
              <input name=\\"title\\" type=\\"text\\" onChange={[Function]} onBlur={[Function]} placeholder=\\"Title\\" value=\\"\\" test-id=\\"input-test\\" className=\\"form-input_input\\" disabled={[undefined]} />
              <div>
                <div className=\\"form-input_validation-marker undefined\\" />
                <div className=\\"form-input_error\\" />
              </div>
            </div>
          </FormInput>
          <FormInput label=\\"Author\\" type=\\"text\\" name=\\"author\\" onChange={[Function]} onBlur={[Function]} value=\\"\\">
            <div className=\\"form-input undefined\\">
              <p className=\\"form-input_label\\">
                Author
              </p>
              <input name=\\"author\\" type=\\"text\\" onChange={[Function]} onBlur={[Function]} placeholder=\\"Author\\" value=\\"\\" test-id=\\"input-test\\" className=\\"form-input_input\\" disabled={[undefined]} />
              <div>
                <div className=\\"form-input_validation-marker undefined\\" />
                <div className=\\"form-input_error\\" />
              </div>
            </div>
          </FormInput>
        </div>
        <div className=\\"add-translation_fields-line\\">
          <FormInput label=\\"Email\\" type=\\"email\\" name=\\"email\\" onChange={[Function]} onBlur={[Function]} value=\\"\\" className=\\"blocked\\" disabled={true}>
            <div className=\\"form-input blocked\\">
              <p className=\\"form-input_label\\">
                Email
              </p>
              <input name=\\"email\\" type=\\"email\\" onChange={[Function]} onBlur={[Function]} placeholder=\\"Email\\" value=\\"\\" test-id=\\"input-test\\" className=\\"form-input_input\\" disabled={true} />
              <div>
                <div className=\\"form-input_validation-marker undefined\\" />
                <div className=\\"form-input_error\\" />
              </div>
            </div>
          </FormInput>
          <FormInput label=\\"Send me a notification via email\\" type=\\"checkbox\\" name=\\"isEmail\\" onChange={[Function]} onBlur={[Function]} value=\\"\\" isValidationDisplay={false}>
            <div className=\\"form-input undefined\\">
              <p className=\\"form-input_label\\">
                Send me a notification via email
              </p>
              <input name=\\"isEmail\\" type=\\"checkbox\\" onChange={[Function]} onBlur={[Function]} placeholder=\\"Send me a notification via email\\" value=\\"\\" test-id=\\"input-test\\" className=\\"form-input_input\\" disabled={[undefined]} />
            </div>
          </FormInput>
        </div>
        <FormTextarea label=\\"Description\\" placeholder=\\"Fill in the description.\\" name=\\"description\\" onChange={[Function]} onBlur={[Function]} value=\\"\\">
          <div className=\\"form-textarea undefined\\">
            <div className=\\"form-textarea_label\\">
              Description
            </div>
            <textarea className=\\"form-textarea_input\\" placeholder=\\"Fill in the description.\\" name=\\"description\\" onChange={[Function]} onBlur={[Function]} value=\\"\\" />
            <div className=\\"form-textarea_error\\" />
          </div>
        </FormTextarea>
        <div className=\\"add-translation_code-displays\\">
          <FormCodeDisplay value=\\"parameter int;\\\\nstorage int;\\\\ncode { CAR ;\\\\n       PUSH int 1 ;\\\\n       ADD ;\\\\n       NIL operation ;\\\\n       PAIR }\\" type=\\"Micheline\\">
            <div className=\\"form-code-display\\">
              <div className=\\"form-code-display_type\\">
                Micheline
              </div>
              <div className=\\"form-code-display_value\\">
                <pre>
                  parameter int;
                  storage int;
                  code { CAR ;
                         PUSH int 1 ;
                         ADD ;
                         NIL operation ;
                         PAIR }
                </pre>
              </div>
            </div>
          </FormCodeDisplay>
          <FormCodeDisplay value=\\"[ { \\"prim\\": \\"parameter\\", \\"args\\": [ { \\"prim\\": \\"int\\" } ] }, { \\"prim\\": \\"storage\\", \\"args\\": [ { \\"prim\\": \\"int\\" } ] }, { \\"prim\\": \\"code\\", \\"args\\": [ [ { \\"prim\\": \\"CAR\\" },{ \\"prim\\": \\"PUSH\\", \\"args\\": [ { \\"prim\\": \\"int\\" }, { \\"int\\": \\"1\\" } ] },{ \\"prim\\": \\"ADD\\" },{ \\"prim\\": \\"NIL\\", \\"args\\": [ { \\"prim\\": \\"operation\\" } ] },{ \\"prim\\": \\"PAIR\\" } ] ] } ]\\\\n\\" type=\\"Michelson\\">
            <div className=\\"form-code-display\\">
              <div className=\\"form-code-display_type\\">
                Michelson
              </div>
              <div className=\\"form-code-display_value\\">
                <pre>
                  [ { &quot;prim&quot;: &quot;parameter&quot;, &quot;args&quot;: [ { &quot;prim&quot;: &quot;int&quot; } ] }, { &quot;prim&quot;: &quot;storage&quot;, &quot;args&quot;: [ { &quot;prim&quot;: &quot;int&quot; } ] }, { &quot;prim&quot;: &quot;code&quot;, &quot;args&quot;: [ [ { &quot;prim&quot;: &quot;CAR&quot; },{ &quot;prim&quot;: &quot;PUSH&quot;, &quot;args&quot;: [ { &quot;prim&quot;: &quot;int&quot; }, { &quot;int&quot;: &quot;1&quot; } ] },{ &quot;prim&quot;: &quot;ADD&quot; },{ &quot;prim&quot;: &quot;NIL&quot;, &quot;args&quot;: [ { &quot;prim&quot;: &quot;operation&quot; } ] },{ &quot;prim&quot;: &quot;PAIR&quot; } ] ] } ]
                  
                </pre>
              </div>
            </div>
          </FormCodeDisplay>
        </div>
        <div className=\\"add-translation_buttons\\">
          <FormButton label=\\"close\\" type=\\"button\\" stylingType=\\"secondary\\" onClick={[Function: onClick]}>
            <button className=\\"form-button form-button_secondary undefined\\" data-testid=\\"form-button\\" disabled={[undefined]} type=\\"button\\" onClick={[Function: onClick]}>
              close
            </button>
          </FormButton>
          <FormButton label=\\"send\\" type=\\"submit\\" disabled={true}>
            <button className=\\"form-button form-button_submit undefined\\" data-testid=\\"form-button\\" disabled={true} type=\\"submit\\" onClick={[undefined]}>
              send
            </button>
          </FormButton>
        </div>
      </form>
    </Formik>
  </div>
</AddTranslation>"
`;
